lib/host/events/log_algo_data.js

```diff
@@ -0,0 +1,58 @@
+'use strict'
+
+const fs = require('fs')
+const path = require('path')
+const _isEmpty = require('lodash/isEmpty')
+const _isBoolean = require('lodash/isBoolean')
+const fileOps = require('../../util/fs_operations')
+
+module.exports = async (aoHost, gid, values) => {
+  const { logAlgoOpts = {}, instances } = aoHost
+  const instance = instances[gid]
+
+  if (!instance) {
+    return
+  }
+
+  const { state = {}, h = {} } = instance
+  const { debug, updateState } = h
+  const { id, headersForLogFile, wls } = state
+
+  const { logAlgo, logAlgoDir } = logAlgoOpts
+
+  if (!_isBoolean(logAlgo) || !logAlgo || _isEmpty(logAlgoDir)) {
+    return
+  }
+
+  const logDir = path.join(logAlgoDir, `${id}_logs`)
+  const filePath = path.join(logDir, `${gid}.csv`)
+
+  if (wls) {
+    fileOps.writeToFile(wls, values)
+    return
+  }
+
+  const dirExists = await fileOps.pathExists(logDir)
+
+  if (!dirExists) {
+    debug('creating logs folder(%s) for %s algo', filePath, id)
+    try {
+      await fileOps.createDir(logDir)
+    } catch (err) {
+      return debug('error creating logs folder(%s) for %s algo %O', filePath, id, err)
```

Comment:
if this case happens, this code is never called:

`await updateState(instance, { wls: writeAlgoLogStream })`

so the next algo order will crash... 

should we abort in general with an error already here and send it to the UI?

Reply:
it wont... kept the algo orders alive even when error was found during logging